tests/backend/integration/api/projects/test_teams.py

```diff
@@ -0,0 +1,238 @@
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response_body["SubCode"], "InvalidToken")
+
+    def test_get_project_teams_for_non_existent_project_fails(self):
+        """
+        Test that endpoint returns 404 when retrieving teams for non-existent projects
+        """
+        response = self.client.get(
+            "/api/v2/projects/99/teams/",
+            headers={"Authorization": self.test_user_session_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(response_body["Error"], "Project Not Found")
+        self.assertEqual(response_body["SubCode"], "NotFound")
+
+    def test_get_project_teams_passes(self):
+        """
+        Test that endpoint returns 200 when an authenticated user retrieves teams
+        """
+        response = self.client.get(
+            self.all_project_teams_url,
+            headers={"Authorization": self.test_user_session_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response_body["teams"]), 0)
+        self.assertEqual(response_body["teams"], [])
+        # setup: add team to project
+        assign_team_to_project(project=self.test_project, team=self.test_team, role=0)
+        response = self.client.get(
+            self.all_project_teams_url,
+            headers={"Authorization": self.test_user_session_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response_body["teams"]), 1)
+        self.assertEqual(response_body["teams"][0]["name"], TEST_TEAM_NAME)
+        self.assertEqual(response_body["teams"][0]["role"], 0)
+
+    # post
+    def test_assign_team_to_project_by_unauthenticated_user_fails(self):
+        """
+        Test that endpoint returns 401 when unauthenticated user assigns team to project
+        """
+        response = self.client.post(
+            self.single_project_team_url, json={"role": TeamRoles.MAPPER.name}
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response_body["SubCode"], "InvalidToken")
+
+    def test_assign_team_to_project_by_non_admin_fails(self):
+        """
+        Test that endpoint returns 403 when non admin assigns team to a project
+        """
+        response = self.client.post(
+            self.single_project_team_url,
+            json={"role": TeamRoles.MAPPER.name},
+            headers={"Authorization": self.test_user_session_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response_body["Error"], "User is not an admin or a manager for the team"
+        )
+        self.assertEqual(response_body["SubCode"], "UserPermissionError")
+
```

Comment:
A test to determine if 403 is raised when the user does not have permission to update the project appears to be missing as well. Only a team's user permission is tested in the test above

Also test to determine whether a 400 is raised when an invalid team role is passed appears to be missing.